adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java

```diff
@@ -23,23 +23,49 @@ public AdapterEndpoint(Adapter adapter) {
     this.adapter = adapter;
   }
 
+
   @PostMapping(value = Mappings.IMPORT_PATH, produces = "application/json")
   public ResponseEntity<?> executeDataImport(
-          @Valid @RequestBody AdapterConfig config,
-          @RequestParam(required = false) boolean includeData) {
+    @Valid @RequestBody AdapterConfig config,
+    @RequestParam(required = false) boolean includeData) {
+    return this.executePreview(config, includeData);
+  }
+
+  @PostMapping(value = Mappings.PREVIEW_PATH, produces = "application/json")
+  public ResponseEntity<?> executePreview(
+    @Valid @RequestBody AdapterConfig config,
+    @RequestParam(required = false, defaultValue = "true") boolean includeData) {
     try {
       DataBlob imported = adapter.executeJob(config);
 
-      if(includeData) {
-        return ResponseEntity.ok(imported.getData());
+      if (includeData) {
+        return ResponseEntity.ok(imported);
+      }
+
+      return ResponseEntity.ok(imported.getMetaData());
+
+    } catch (ResponseStatusException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+  }
+
+  @PostMapping(value = Mappings.RAW_PREVIEW_PATH, produces = "application/json")
+  public ResponseEntity<?> executeRawPreview(
+    @Valid @RequestBody ProtocolConfig config,
+    @RequestParam(required = false, defaultValue = "true") boolean includeData) {
+    try {
+      DataBlob imported = adapter.executeRawImport(config, includeData);
```

Comment:
2nd parameter here called `ìncludeData` and in method called `persistData`. Aren't those two different things? Are we mixing things up? Or is it two names for the same thing?

Reply:
IncludeData implies that it does not need to persisted. However, I should probably remove that parameter altogether since in a non-raw import requests with includeData param are persisted as well. 
Maybe we should prevent persistence in both cases, but this should probably be done in another PR, what do you think?

Reply:
Yeah move to other PR. Maybe remove that parameter for the raw preview endpoint completly in this PR?